int32 blue_angle
int32 green_angle
int32 blue_x
int32 blue_y
int32 green_x
int32 green_y
